jcl/src/java.base/share/classes/openj9/tools/attach/diagnostics/base/DiagnosticProperties.java

```diff
@@ -0,0 +1,256 @@
+		baseProperties.setProperty(key, Long.toString(value));
+	}
+
+	/**
+	 * Add a property with a boolean value
+	 * @param key property name
+	 * @param value property value
+	 */
+	public void put(String key, boolean value) {
+		baseProperties.setProperty(key, Boolean.toString(value));
+	}
+
+	private void checkExists(String key) throws IOException {
+		if (!containsField(key)) {
+			throw new IOException("key " + key + " not found"); //$NON-NLS-1$ //$NON-NLS-2$
+		}
+	}
+
+	/**
+	 * Test if the given property is present
+	 * @param key property name
+	 * @return true if the property name is found
+	 */
+	public boolean containsField(String key) {
+		return baseProperties.containsKey(key);
+	}
+
+	/**
+	 * Retrieve the value of a property and convert it to an int.
+	 * @param key name of the property
+	 * @return value of the property as an int
+	 * @throws NumberFormatException if the value is not a string representing a number.
+	 * @throws IOException  if the property is missing
+	 */
+	public int getInt(String key) throws NumberFormatException, IOException {
+		checkExists(key);
+		return Integer.parseInt(baseProperties.getProperty(key));
+	}
+
+	/**
+	 * Retrieve the value of a property and convert it to an long.
+	 * @param key name of the property
+	 * @return value of the property as a long
+	 * @throws NumberFormatException if the value is not a string representing a number.
+	 * @throws IOException  if the property is missing
+	 */
+	public long getLong(String key) throws NumberFormatException, IOException  {
+		checkExists(key);
+		return Long.parseLong(baseProperties.getProperty(key));
+	}
+
+	/**
+	 * Retrieve the value of a property and convert it to a boolean
+	 * @param key name of the property
+	 * @return value of the property as a boolean
+	 * @throws IOException  if the property is missing
+	 */
+	public boolean getBoolean(String key) throws IOException { 
+		checkExists(key);
+		return Boolean.parseBoolean(baseProperties.getProperty(key));
+	}
+
+	/**
+	 * Return a property value for the given key
+	 * @param key property name
+	 * @return property value or null if the property is not found
+	 */
+	public String getProperty(String key) {
+		return baseProperties.getProperty(key);
+	}
+
+	/**
+	 * Retrieve a value of a simple type, i.e. boxed scalar type, from a specified property
+	 * @param typeName Fully qualified name of the type
+	 * @param key Name of the property containing the value
+	 * @return object containing the value
+	 * @throws NumberFormatException  if the property is malformed
+	 */
+	public Object getSimple(String typeName, String key) throws NumberFormatException {
+		Object value = null;
+		String valueString = baseProperties.getProperty(key);
+		if (null != valueString) {
+			switch (typeName) {
+			case "java.lang.Boolean": //$NON-NLS-1$
+				value = Boolean.valueOf(valueString);
+				break;
+			case "java.lang.Character": //$NON-NLS-1$
+				value = Character.valueOf(valueString.charAt(0));
```

Comment:
This might throw `StringIndexOutOfBoundsException`.

Reply:
Right.  Checking for empty strings.

Reply:
Now this doesn't support an empty `java.lang.String`: might that be a problem?